iothub/device/samples/getting started/RoutingTutorial/SimulatedDevice/Program.cs

```diff
@@ -0,0 +1,189 @@
+    /// This program encodes the message body so it can be queried against by the Iot hub.
+    ///
+    /// If you want to read an encoded message body, you can do this:
+    ///   * route the message to storage,
+    ///   * retrieve the message from the storage account by downloading the blob,
+    ///   * use the ReadOneRowFromFile method at the bottom of this module to read the first row of the file.
+    ///   * It will read the file, decode the first row in the file, and write it out to a new file
+    ///       in ASCII so you can view it.
+    /// </summary>
+    internal class Program
+    {
+        private static async Task Main(string[] args)
+        {
+            // Parse application parameters
+            Parameters parameters = null;
+            ParserResult<Parameters> result = Parser.Default.ParseArguments<Parameters>(args)
+                .WithParsed(parsedParams =>
+                {
+                    parameters = parsedParams;
+                })
+                .WithNotParsed(errors =>
+                {
+                    Environment.Exit(1);
+                });
+
+            // If this is false, it will submit messages to the iot hub.
+            // If this is true, it will read one of the output files and convert it to ASCII.
+            if (parameters.ReadTheFile)
+            {
+                // If you want to decode an output file, put the path in ReadOneRowFromFile(),
+                //   uncomment the call here and the return command, then run this application.
+                ReadOneRowFromFile(parameters.FilePath);
+            }
+            else
+            {
+                // Send messages to the simulated device. Each message will contain a randomly generated
+                //   Temperature and Humidity.
+                // The "level" of each message is set randomly to "storage", "critical", or "normal".
+                // The messages are routed to different endpoints depending on the level, temperature, and humidity.
+                //  This is set in the tutorial that goes with this sample:
+                //  http://docs.microsoft.com/azure/iot-hub/tutorial-routing
+
+                Console.WriteLine("Routing Tutorial: Simulated device\n");
+                using var s_deviceClient = DeviceClient.CreateFromConnectionString(parameters.PrimaryConnectionString);
+
+                using var cts = new CancellationTokenSource();
+                Console.CancelKeyPress += (sender, eventArgs) =>
+                {
+                    eventArgs.Cancel = true;
+                    cts.Cancel();
+                    Console.WriteLine("Sample execution cancellation requested; will exit.");
+                };
+
+                Task messages = SendDeviceToCloudMessagesAsync(s_deviceClient, cts.Token);
+
+                Console.WriteLine($"Press Control+C at any time to quit the sample.");
+                await s_deviceClient.CloseAsync(cts.Token);
+                await messages;
```

Comment:
Reorder to make it simpler.

Don't close the client before control-c.

```suggestion
                Console.WriteLine($"Press Control+C at any time to quit the sample.");
                await SendDeviceToCloudMessagesAsync(s_deviceClient, cts.Token);
                await deviceClient.CloseAsync();
```